File: codewars/6kyu/find-the-unique-number/dduck-find-the-unique-number.js
// Find the unique number
// https://www.codewars.com/kata/585d7d5adb20cf33cb000235/

const findUniq = (arr) => arr.map((v) => arr.indexOf(v) === arr.lastIndexOf(v) ? v : undefined)
  .filter((v) => v !== undefined)[0];

console.log(findUniq([0, 1, 0]), 1);
console.log(findUniq([1, 1, 1, 2, 1, 1]), 2);
console.log(findUniq([3, 10, 3, 3, 3]), 10);

// const findUniq = (arr) => {
//   const map = new Map([...new Set(arr)].map((v) => [v, 0]));
//   arr.forEach((v) => {
//     map.set(v, map.get(v) + 1);
//   });
//
//   let uniqueNum = null;
//   [...map.entries()]
//     .forEach((v) => {
//       if (v[1] === 1) {
//         uniqueNum = v[0];
//       }
//     });
//
//   return uniqueNum;
// };

File: codewars/6kyu/find-the-unique-number/dduck-find-the-unique-number.py
# Find the unique number
# https://www.codewars.com/kata/585d7d5adb20cf33cb000235/

from unittest import TestCase


def find_uniq(arr):
    reversed_arr = arr[::-1]
    arr_len = len(arr)

    for v in arr:
        if arr.index(v) == (arr_len - reversed_arr.index(v) - 1):
            return v


TestCase().assertEqual(find_uniq([1, 1, 1, 2, 1, 1]), 2)
TestCase().assertEqual(find_uniq([0, 0, 0.55, 0, 0]), 0.55)
TestCase().assertEqual(find_uniq([3, 10, 3, 3, 3]), 10)
